Source/WebCore/html/HTMLAnchorElement.cpp

```diff
@@ -115,47 +116,18 @@ bool HTMLAnchorElement::isInteractiveContent() const
     return isLink();
 }
 
-static bool hasNonEmptyBox(RenderBoxModelObject* renderer)
```

Comment:
I don't think we should be removing this check.

Reply:
It resolves issue around `https://bugs.webkit.org/show_bug.cgi?id=31496#c7` and taking that bug as example (JSFiddle) - https://jsfiddle.net/b058spv4/, in local build, using tab, now I am able to focus both `float` and `non-float` while in Safari 17.4, I am not.

Reply:
https://github.com/WebKit/WebKit/assets/52317531/1caa6cc2-9866-4737-b339-2b23d3c0a1fb

Reply:
We don't have this concept in `SVGAElement` - just for my learning - any idea why? https://searchfox.org/wubkat/rev/e80ab9289841484d684b81a0ae84ad3cbd793b05/Source/WebCore/svg/SVGAElement.cpp#181

Reply:
@rniwa - Thanks for your guidance on Slack, I retained `hasNonEmptyBox` for time being and rebased this patched and tested all new tests to ensure that we don't regress anything and just pushed new commit. After this lands, I will remove `hasNonEmptyBox` with test case (will work on it). So we can align with all other browsers and have another issue down. Thanks again for your help as always.